lib/src/base_client.dart

```diff
@@ -68,12 +88,12 @@ abstract class BaseClient implements Client {
   /// later point, or it could already be closed when it's returned. Any
   /// internal HTTP errors should be wrapped as [ClientException]s.
   @override
-  Future<StreamedResponse> send(BaseRequest request);
+  Future<StreamedResponse> send(BaseRequest request, {Duration? timeout});
 
   /// Sends a non-streaming [Request] and returns a non-streaming [Response].
   Future<Response> _sendUnstreamed(
       String method, Uri url, Map<String, String>? headers,
-      [body, Encoding? encoding]) async {
+      {dynamic body, Encoding? encoding, Duration? timeout}) async {
```

Comment:
Personally, I usually make all parameters required for internal functions.
It's not great API, but it's incredibly simple and efficient.

Reply:
Sure, this library is small enough that it isn't much of a readability hit.